__tests__/postService.test.js

```diff
@@ -33,21 +34,27 @@ beforeAll(() => {
         };
     });
     postDAO.getPost.mockImplementation(async (id) => {
-        for (let i = 0; i < mockDatabase.length; i++){
-            if (mockDatabase[i].itemID == id){
+        for (let i = 0; i < mockDatabase.length; i++) {
+            if (mockDatabase[i].itemID == id) {
                 return {
                     $metadata: {
                         httpStatusCode: 200
                     },
-                    Item: mockPost1
+                    Item: mockDatabase[i]
+                };
+            } else {
+                return {
+                    $metadata: {
+                        httpStatusCode: 400
+                    }
                 };
             }
         }
         return false;
     });
     postDAO.sendReply.mockImplementation(async (reply, id) => {
-        for (let i = 0; i < mockDatabase.length; i++){
-            if (mockDatabase[i].itemID == id){
+        for (let i = 0; i < mockDatabase.length; i++) {
+            if (mockDatabase[i].itemID == id) {
                 mockDatabase[i].replies.push(reply);
```

Comment:
Something I noticed while testing is that the variable reply is an array instead of an object (not exactly sure why but I'm assuming this is necessary when uploading a new reply to the dynamodb) so I would push reply[0] instead of just reply.